NitroxPatcher/Patches/Dynamic/Planter_ResetStorage_Patch.cs

```diff
@@ -24,12 +25,54 @@ public static void Prefix(Planter __instance)
 
         // This is called from WaterPark.Update so we have no proper way of suppressing it.
         // Thus we restrict to simulation ownership
-        if (!__instance.TryGetIdOrWarn(out NitroxId planterId) ||
-            !Resolve<SimulationOwnership>().HasAnyLockType(planterId))
+        if (__instance.TryGetIdOrWarn(out NitroxId planterId) &&
+            TryGetOwnerNitroxId(__instance, out NitroxId ownerNitroxId) &&
+            Resolve<SimulationOwnership>().HasAnyLockType(ownerNitroxId))
         {
-            return;
+            Resolve<IPacketSender>().Send(new ClearPlanter(planterId));
+        }
+    }
+
+    /// <summary>
+    /// Returns <c>true</c> with <paramref name="ownerNitroxId"/> being the NitroxId of the entity responsible for the planter if it can be found.
+    /// Else returns <c>false</c>.
+    /// </summary>
+    public static bool TryGetOwnerNitroxId(Planter planter, out NitroxId ownerNitroxId)
+    {
```

Comment:
Maybe it should be moved to our model layer, the coupling between Planter_ResetStorage_Patch & FruitPlant_Patch doesn't really make much sense to me

Reply:
I think I would probably agree here. Maybe moving to the NitroxModel Helper area might be a good place